### PR TITLE
Revert "Revert "Fix 388""

### PR DESCRIPTION
**Pull request by nderguti**
Reverts celements/celements-blog#20

https://synjira.atlassian.net/browse/CELDEV-388

### CODE REVIEW

**Review by msladek — CHANGES_REQUESTED** [review]
project has compile errors, please finish changes first before requesting reviews.

i think these errors came in due to a bad merge with the dev

File: component/pom.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
     <dependency>
       <groupId>com.celements</groupId>
       <artifactId>celements-core</artifactId>
-      <version>1.147</version>
+      <version>1.160</version>
```
Comment (msladek) [inline component/pom.xml]: update to current version 1.162

File: component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java [hunk 2]
```diff
@@ -414,89 +416,96 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
...
     if ((blogObj != null) && (blogObj.getIntValue("unsubscribe_info") == 1)) {
-      unsubscribeLink = blogDocument.getExternalURL("view",
-          "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe" + "&emailadresse="
-              + emailAddresse, context);
+      try {
+        unsubscribeLink = blogDocument.getExternalURL("view",
+            "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe&emailadresse="
+                + URLEncoder.encode(emailAddresse, StandardCharsets.UTF_8.name()), getContext());
+      } catch (UnsupportedEncodingException e) {
+        LOGGER.error("UTF-8 unsupported.");
```
Comment (msladek) [inline component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java]: give exception a better name and add it to error logging

File: component/src/test/java/com/celements/blog/plugin/NewsletterReceiversTest.java [hunk 3]
```diff
@@ -19,18 +19,67 @@
...
   private NewsletterReceivers comp;
 
   @Before
-  public void setUp_NewsletterReceiversTest() throws Exception {
-    // comp = new NewsletterReceivers(, getContext());
+  public void prepareTest() throws Exception {
+    comp = new NewsletterReceivers();
+  }
+
+  public void unsubscribeLink(String email) throws Exception {
```
Comment (msladek) [inline component/src/test/java/com/celements/blog/plugin/NewsletterReceiversTest.java]: private suffices

**Review by msladek — CHANGES_REQUESTED** [review]

File: component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java [hunk 4]
```diff
@@ -414,89 +416,96 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
...
+          getContext());
+      LOGGER.debug("context=" + getContext());
       vcontext.put("msg", msgTool);
       vcontext.put("adminMsg", msgTool);
       header += renderCommand.renderDocument(headerRef, renderLang);
       LOGGER.debug("Additional header rendered.");
     } else {
       LOGGER.debug("No additional header. Doc does not exist: " + headerRef);
     }
+
```
Comment (ebeutler) [inline component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java]: please remove empty line

File: component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java [hunk 5]
```diff
@@ -414,89 +416,96 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
...
     LOGGER.debug("rendering content in " + renderLang);
-    context.setLanguage(renderLang);
+
```
Comment (ebeutler) [inline component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java]: please remove empty line

**Review by ebeutler — CHANGES_REQUESTED** [review]
Just two empty lines to remove.

**Review by msladek — APPROVED** [review]

**Review by ebeutler — APPROVED** [review]

**Pull request merged**